test/take.cc

```diff
@@ -159,5 +159,37 @@ TEST(Take, UniquePtr) {
   EXPECT_EQ(2, *result[1]);
 }
 
+TEST(Take, Stream1) {
+  auto s = []() {
+    return Stream<int>()
+               .next([i = 0](auto& k) mutable {
+                 if (i < 2) {
+                   k.Emit(i++);
+                 }
+               })
+        | TakeRange(0, 2)
+        | Collect<std::vector<int>>();
+  };
+
+  EXPECT_THAT(*s(), ElementsAre(0, 1));
+}
+
+TEST(Take, Stream2) {
```

Comment:
Given that the previous test passes that implies that this doesn't test anything! So let's just remove it please.